comprehensive_microservice/coder.py

```diff
@@ -7,8 +8,24 @@ class DataModel(BaseModel):
 
 
 class DataCoder:
+    def __init__(self):
+        self.storage = {}
+
     def save(self, data: DataModel) -> str:
-        raise NotImplementedError
+        while True:
+            new_id = str(randint(1, 100000))
+            if new_id not in self.storage.keys():
+                break
+        self.storage[new_id] = (data.title, data.text)
```

Comment:
No need in saving it as tuple, let's save `data` object directly:


```suggestion
        self.storage[new_id] = data
```